### DISCUSSION TITLE
Is it possible to override the background colour of the Underline control on validation error?

### DISCUSSION THREAD

**Discussion by LeeDawsonUK** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3469] [Q&A]
I'm trying to override the colour of a TextBox's Underline when there is a validation error on the control. The use case here is that our application has different severities of validation error and we want to display some in red and some in a less serious amber.

This is what I have tried:

```
 <Style x:Key="MyMaterialDesignTextBox" BasedOn="{StaticResource MaterialDesignTextBox}" TargetType="{x:Type TextBox}">
     <Style.Triggers>
         <Trigger Property="Validation.HasError" Value="true">
             <Setter Property="BorderBrush" Value="Orange" />
             <Setter Property="materialDesign:TextFieldAssist.UnderlineBrush" Value="Orange"/>
         </Trigger>
     </Style.Triggers>
 </Style>
```

The setting of the `BorderBrush` works, but that's only visible when the control loses focus. The setting of the `UnderlineBrush` has no effect, it remains the standard validation error brush. I think this is because the `Validation.HasError` trigger has hard-coded `MaterialDesign.Brush.ValidationError` (i.e. the UnderlineBrush is irrelevant in this scenario):

```
MaterialDesignTheme.TextBox.xaml:

<Trigger Property="Validation.HasError" Value="true">
   <Setter Property="BorderBrush" Value="{DynamicResource MaterialDesign.Brush.ValidationError}" />
   <Setter TargetName="Underline" Property="Background" Value="{DynamicResource MaterialDesign.Brush.ValidationError}" />
</Trigger>
```

I cannot override this in a derived Style because referencing elements in the template via `TargetName` is not allowed on a Style setter.

Is what I'm trying to do here possible without making my own copy of the TextBox template?

**Comment by nicolaihenriksen** [comment]
@LeeDawsonUK Not tested, but off the top of my head I would say you could simply hijack the resource (key) for that `TextBox` then. Of course you will have a task ahead if you support theming in your app. Something along these lines:

```xaml
<Style x:Key="MyMaterialDesignTextBox" BasedOn="{StaticResource MaterialDesignTextBox}" TargetType="{x:Type TextBox}">
  <Style.Resources>
    <SolidColorBrush x:Key="MaterialDesign.Brush.ValidationError" Color="Orange" po:Freeze="True" />
  </Style.Resources>
   <Style.Triggers>
       <Trigger Property="Validation.HasError" Value="true">
           <Setter Property="BorderBrush" Value="Orange" />
       </Trigger>
   </Style.Triggers>
</Style>
```
